🚩: Final Milestone: React Capstone Project

Ruthmy opened the pull request:
# 🚩: Final Milestone: React Capstone Project - `React` `Redux Toolkit`

## Hi there! @Ruthmy here.

In this project I have developed a Weather Metrics App using the Current Conditions API from AccuWeather. The app is called `Weather App` and it is a single page application (SPA) that allows users to check various weather metrics for different locations.
- The app has been deployed to `GitHub Pages` and you can find it [here](https://ruthmy.github.io/weather-app/).
- To get the data for the weather, I have used the [Current Conditions API](https://developer.accuweather.com/accuweather-current-conditions-api/apis) from `AccuWeather`.

---

## **Website Goals:**

The goals of this project are to:

- Create a mobile web application to display weather metrics.
- Implement a user-friendly interface with a list of weather metrics.
- Allow users to filter weather metrics by location.
- Provide detailed information for each weather metric.
- Apply a visually appealing design based on the provided [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)).

---

## **Project Goals:**

- [x] The project is developed as a single-page application (SPA) using React and Redux.
- [x] Data retrieved from the AccuWeather API is stored in the Redux store.
- [x] Utilize React Router for efficient navigation.
- [x] Each page has a unique route within the SPA.
- [x] Implement React components to create a seamless user experience.
- [x] Integrate Redux to manage state and data retrieval from the API.
- [x] Write unit tests have been implemented to ensure the functionality and reliability of the application.
- [x] Deploy the application online using `GitHub Pages`.
- [x] Record a 5-minute video demonstrating the app's features and code highlights.
- [x] Follow the [GitFlow as a workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) for version control.
- [x] Follow the best practices for [HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- [x] Documented the work in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

<br>

---

[Presentation Video - JavaScript Capstone](https://drive.google.com/file/d/1-eSDabquIJCbvbQblqBPvSwbDIvaRfIb/view)

---

<br>

### Thank you for taking the time to review this PR ⭐!

<br>

---

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
<br>

Ruthmy committed:
Created a base install using create-react-app

Ruthmy committed:
Install the React Redux linters

Ruthmy committed:
Modified 'package.json' -file- install the react-router dom 6

Ruthmy committed:
Modified 'App.js' -component- change extension to jsx

Ruthmy committed:
Modified 'index.js' -component- change extension to jsx

Ruthmy committed:
Modified 'index.css' -stylesheet- updated the basic style for the project

Ruthmy committed:
Modified 'index.html' -file- Added the fonts and deleted the unnecesary comments

Ruthmy committed:
Modified 'package.json' -file- Install the redux library

Ruthmy committed:
Added 'Navbar.jsx' and 'Navbar.css' reused component from another project

Ruthmy committed:
Modified 'package.json' -file- Install axios, jest and redux mocks and necessary test dependencies

Ruthmy committed:
Created 'store.js' -Redux file- Redux store for weather app project

Ruthmy committed:
Created 'weatherSlice.js' -State Slice- to manage the state for the weather functionality

Ruthmy committed:
Modified 'index.jsx' -component- Added the redux store and provider

Ruthmy committed:
Modified 'Navbar.css' -stylesheet- Deleted the old styles

Ruthmy committed:
Deleted 'reportWebVitals.js' -file- as an unnecesary file for the project

Ruthmy committed:
Modified 'App.css' -stylesheet- Deleted unnecessary styles and selector.

Ruthmy committed:
Merge pull request #1 from Ruthmy/feature/basicSetUp

🚩: Milestone 1: Weather App | feature/basicSetUp

Ruthmy committed:
Created 'LocationSlice.jsx' -component- to fetch the continents from the Location API.

Ruthmy committed:
Modified 'index.jsx' -component- Added the dispatch for the fetchLocation thunk from the LocationSlice.

Ruthmy committed:
Created 'weatherAPI.js' -file- to manage the key and urls from the APIs

Ruthmy committed:
Modified 'LocationSlice.js' -State Slice- improved the API call and extraReducers

Ruthmy committed:
Created 'weatherSlice.js' -State Slice- Slice for the current weather API

Ruthmy committed:
Modified 'store.js' -redux store- added the Reducers from the Slices

Ruthmy committed:
Modified 'Navbar.jsx' -component- Updated the Navbar default content

Ruthmy committed:
Created 'Home.jsx' -component- Main component for the Main Page

Ruthmy committed:
Created 'CountryDetails.jsx' -component- to show the weather country details.

Ruthmy committed:
Modified 'App.jsx' -component- Updated the routes to math with the useParams Hook.

Ruthmy committed:
Created 'Continent.jsx' -component- to list the countries in a specific continent

Ruthmy committed:
Modified 'App.jsx' -component- Fixed a typo with the Continent component route

Ruthmy committed:
Created 'Home.css' 'Continent.css' and 'Navbar.css' -stysheets- for its respective components. Updated the 'Continent.jsx' import.

Ruthmy committed:
Modified 'index.css' -stylesheet- Added new selector/properties to help the default styles

Ruthmy committed:
Added all the svg images for the continents

Ruthmy committed:
Created 'continentsImages.js' -file- Object with all the continent images

Ruthmy committed:
Modified 'Asia' and 'Antarctica' -svg - Deleted the background

Ruthmy committed:
Created 'regionList.js' -file- as the offline info for continents list

Ruthmy committed:
Modified 'LocationSlice.js' -State Slice- Updated handling of reject status of fetchLocation promise

Ruthmy committed:
Modified 'Home.jsx' -component- updated handling of the API data

Ruthmy committed:
Merge pull request #2 from Ruthmy/feature/fetchAPI

🚩: Milestone 2: Weather App | feature/fetchAPI

Ruthmy committed:
Modified 'index.html' updated the font imports.

Ruthmy committed:
Added new svg for the buttons/icons and updated America svg

Ruthmy committed:
Modified 'continentsImages.js' -file- changed the America continent name, remove an extra s

Ruthmy committed:
Modified 'regionList.js' -file- Deleted the regions without countries to list

Ruthmy committed:
Modified 'Home.jsx' -component- Added the app logo, modified the map function that render the continents

Ruthmy committed:
Modified 'Home.css' -stylesheet- Added the mobile styles for the Home component

Ruthmy committed:
Modified 'index.jsx' -component- Removed the fetchLocation that hidrates the store

Ruthmy committed:
Modified 'index.css' -stylesheet- Added the new properties/selector needed to match the original design

Ruthmy committed:
Modified 'Navbar.jsx' -component- Updated the navbar buttons with its icons

Ruthmy committed:
Modified 'Navbar.css' -stylesheet- Added new selectors to match the original design

Ruthmy committed:
Modified 'weatherAPI.js' -file- Updated the current conditions URL, deleted unused URLs

Ruthmy committed:
Modified 'Home.jsx' -component- Fixed a tipo in the Logo img alt

Ruthmy committed:
Merge pull request #3 from Ruthmy/feature/HomePage

🚩: Milestone 3: Weather App | feature/homePage

Ruthmy committed:
Modified 'Home.jsx' -component- to send the Continent component the LocalizedName instead of the EnglishName

Ruthmy committed:
Deleted 'LocationSlice.js' -State Slice- This Slice belongs to an API that is no longer required.

Ruthmy committed:
Modified 'weatherSlice.js' -State Slice- Improved the thunk and extraReducers functionality

Ruthmy committed:
Modified 'store.js' -Redux Store- deleted unnecessary reducers.

Ruthmy committed:
Modified 'index.jsx' -component- Added the fethWeather thunk to hidrate the store.

Ruthmy committed:
Modified 'Continent.jsx' -component- Setted the state and added a filter funtion to shows the countries from the actual continent

Ruthmy committed:
Modified 'index.css' - Deleted unnecesaary selectors

Ruthmy committed:
Modified 'Continent.css' -stylesheet- Added the necessary selectors to match the original design

Ruthmy committed:
Merge pull request #4 from Ruthmy/feature/continentPage

🚩: Milestone 4: Weather App | feature/continentPage

Ruthmy committed:
Change name from 'CountryDetails.jsx' to 'CityDetails.jsx' -component- Added all the necesary logic and structure

Ruthmy committed:
Modified 'App.jsx' -component- Updated the CityDetails component for name change

Ruthmy committed:
Modified 'Continent.jsx' -component- Improved the way to get info from state and change some class names

Ruthmy committed:
Modified 'Navbar.jsx' -component- Deleted unnecessary classes

Ruthmy committed:
Created 'weatherIcons.js' -file- as an object container for all the weather related images

Ruthmy committed:
Created 'CityDetails.css' -stylesheet- for the CityDetails component

Ruthmy committed:
Modified 'Continent.css' -stylesheet- Updated the class names

Ruthmy committed:
Modified 'Home.css' -stylesheet- moved 4 selectors to the index

Ruthmy committed:
Modified 'index.css' -stylesheet- Added 4 selectors that affect all components

Ruthmy committed:
Merge pull request #5 from Ruthmy/feature/cityDetailsPage

🚩: Milestone 5: Weather App | feature/cityDetailsPage

Ruthmy committed:
Added a new arrow image in the Navbar component

Ruthmy committed:
Modified 'Home.jsx' -component- Improved the use of the useSelector hook

Ruthmy committed:
Created the tests for the 'App.jsx' component

Ruthmy committed:
Created the tests for the 'Navbar.jsx' component

Ruthmy committed:
Created the tests for the 'Home.jsx' component

Ruthmy committed:
Modified 'index.css' - Added the p.continents__cities selector

Ruthmy committed:
Added the test for the Continent component

Ruthmy committed:
Modified 'CityDetails.jsx' -component-  Change the alt atribute in the weather icon to match the current weatehr icon name

Ruthmy committed:
Added tests for the CityDetails component

Ruthmy committed:
Modified 'weatherSlice.js' -State Slice- Added an export in the initialState const

Ruthmy committed:
Created the tests for the weatherSlice state

Ruthmy committed:
Modified 'weatherAPI.js' - Updated the apikey because the requests has been exceeded

Ruthmy committed:
Updated the 'packege.json' -file- to install the github pages

Ruthmy committed:
Modified 'README.md' -file- Updated the Live Demo and other info details

Ruthmy committed:
Merge pull request #6 from Ruthmy/testing/testComponents

🚩: Milestone 6: Weather App | testing/testComponents

topeogunleye commented on src/components/Home.jsx:
```diff
@@ -0,0 +1,60 @@
+import logo from '../assets/img/logo.svg';
+import '../styles/Home.css';
+
+const Home = () => {
```
- [ ] As required by the project requirement, the Home page should have a list of categories that could be filtered using given parameters, e.g. category name. 

For Example: You could have a search bar to search for continents by name

topeogunleye reviewed (commented):
Hi @Ruthmy,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Git flow Not Used ⚠️

- Pull request is to be made from the feature branch into the develop branch. I understand that on the last block, you may have used this flow for your group projects. However because this week you are working solo and not making any code reviews, you can use a normal git flow. Please see the project requirement [link here](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Make%20sure%20that%20you%20used%20correct%20Gitflow)


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

Ruthmy committed:
Modified 'regionListArray.js' -file- change name to continentsList.js

Ruthmy committed:
Modified 'Home.jsx' -component- Added a useState hook and an input to search by continent

Ruthmy committed:
Modified 'index.css' and 'Home.css' -stylesheets- Added the necessary styles for the input.

Ruthmy committed:
Merge pull request #8 from Ruthmy/testing/testComponents

Testing/test components

Ruthmy commented:
Hi @topeogunleye  I really appreciate your time in reviewing my code and the feedback you have given me.

Even so I need to add that, I already implemented [this changes you asked for first](https://prnt.sc/lbMUd73EHnnD), to include a filter or search by categories. So I implemented a search bar.
But almost an hour later you invalidate the review by GitFow.

Anyway, I reread the project requirements and the [GitFlow guide](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), and yes, my PR is the merge from dev to main.
I took a few minutes to check the profiles of other colleagues, [for example this one](https://github.com/grauJavier/react-capstone-project/pull/6), and they are approved and under the same workflow, several feature branches with merge in dev and at the end with a merge at the end from dev to main.

I would be very grateful if you could please tell me what happened.

tugsem reviewed (approved):
## STATUS: APPROVED ✔️ 
Hi @Ruthmy , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

![Alt Text](https://media.giphy.com/media/3o6Ztp6SgDdzR0XD2w/giphy.gif)
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

topeogunleye commented:
> But almost



> Hi @topeogunleye I really appreciate your time in reviewing my code and the feedback you have given me.
> 
> Even so I need to add that, I already implemented [this changes you asked for first](https://prnt.sc/lbMUd73EHnnD), to include a filter or search by categories. So I implemented a search bar. But almost an hour later you invalidate the review by GitFow.
> 
> Anyway, I reread the project requirements and the [GitFlow guide](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), and yes, my PR is the merge from dev to main. I took a few minutes to check the profiles of other colleagues, [for example this one](https://github.com/grauJavier/react-capstone-project/pull/6), and they are approved and under the same workflow, several feature branches with merge in dev and at the end with a merge at the end from dev to main.
> 
> I would be very grateful if you could please tell me what happened.

@Ruthmy I asked my Senior Code Reviewers and they suggested that I invalidate. Congratulations on the approval. Congratulations on your Approval 🎊
